tests/v2/validation/publicapi/clusterroletemplatebindings/crtb_create_steve_test.go

```diff
@@ -0,0 +1,292 @@
+	inputData[roleTemplateName] = "roleTemplateNameNotExisted"
+
+	_, err = client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.Error(t, err)
+	require.ErrorContains(t, err, badRequest)
+}
+
+func validateErrorBadRequestByPassingDifferentClusterAndNameSpace(t *testing.T, client *rancher.Client, clusterID string, newUser *management.User, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(t, err)
+	crtbName := namegen.RandStringLower(crtbNameWithTenChar)
+
+	inputData := unmarshalCrtbJsonTemplate(t, readJson, clusterID, crtbName, newUser.ID, defaultRoleTemplateName)
+	inputData[clusterName] = localCluster
+
+	_, err = client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.Error(t, err)
+	require.ErrorContains(t, err, badRequest)
+}
+
+func validateErrorNameSpaceNameNotExisted(t *testing.T, client *rancher.Client, clusterID string, newUser *management.User, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(t, err)
+	crtbName := namegen.RandStringLower(crtbNameWithTenChar)
+
+	inputData := unmarshalCrtbJsonTemplate(t, readJson, clusterID, crtbName, newUser.ID, defaultRoleTemplateName)
+	inputData[metadata].(map[string]interface{})[namespace] = "namespacenotexisted"
+	inputData[clusterName] = "namespacenotexisted"
+
+	_, err = client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.Error(t, err)
+	require.ErrorContains(t, err, notFound)
+}
+
+func validateErrorClusterNameAndNameSpaceNameDifferent(t *testing.T, client *rancher.Client, clusterID string, newUser *management.User, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(t, err)
+	crtbName := namegen.RandStringLower(crtbNameWithTenChar)
+
+	inputData := unmarshalCrtbJsonTemplate(t, readJson, clusterID, crtbName, newUser.ID, defaultRoleTemplateName)
+	inputData[clusterName] = localCluster
+
+	_, err = client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.Error(t, err)
+	require.ErrorContains(t, err, cluserNameAndNameSpaceSameValue)
+}
+
+func (crtb *CRTBGenTestSuite) TestCreateCrtbWithAllRequiredInputValues() {
+	tests := []struct {
+		roleTemplateId string
+	}{
+		{clusterOwner},
+		{clusterMember},
+		{crtbManage},
+	}
+	for _, tt := range tests {
+		crtb.Run("Verify that the creation of crtb is successful with roleTemplateName "+tt.roleTemplateId, func() {
+			createCrtbWithAllRequiredInputValues(crtb.T(), crtb.client, crtb.clusterID, crtb.newUser, tt.roleTemplateId, crtb.crtbJsonTemplateData)
+		})
+	}
+}
+
+func (crtb *CRTBGenTestSuite) TestCreateCrtbWithUserPrincipalName() {
+	createCrtbWithUserPrincipalNameAsInput(crtb.T(), crtb.client, crtb.clusterID, crtb.newUser, crtb.crtbJsonTemplateData)
+}
+
+func (crtb *CRTBGenTestSuite) TestCreateCrtbWithServiceAccountAsInput() {
```

Comment:
noticed that various cluster members - clusterOwner, clusterMember and crtbManage are only added in the `TestCreateCrtbWithAllRequiredInputValues`test. Is there a specific reason why we are not verifying the other tests with different cluster permissions?